packages/swapper/src/swappers/ThorchainSwapper/evm/utils/getThorTxData.ts

```diff
@@ -33,8 +34,8 @@ export const getThorTxInfo = async ({
   if (maybeInboundAddress.isErr()) throw maybeInboundAddress.unwrapErr()
   const inboundAddress = maybeInboundAddress.unwrap()
 
-  const router = inboundAddress.router
-  const vault = inboundAddress.address
+  const router = getAddress(inboundAddress.router as string)
```

Comment:
The below might be better so we rely on `getAddress`'s error handling instead of of getting a runtime error if `router` actually is `undefined`.

```suggestion
  const router = getAddress(inboundAddress.router ?? '')
```